bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/ObserveUtils.java

```diff
@@ -195,9 +195,14 @@ public static void startCallableObservation(Environment env, boolean isRemote, b
         } else {
             String className = typeDef.getClass().getCanonicalName();
             String[] classNameSplit = className.split("\\.");
-            int lastIndexOfDollar = classNameSplit[3].lastIndexOf('$');
-            newObContext.setObjectName(classNameSplit[0] + "/" + classNameSplit[1] + "/"
-                    + classNameSplit[3].substring(lastIndexOfDollar + 1));
+            if (classNameSplit.length == 4) {   // Object from a project module
```

Comment:
is it possible to use ,  `typeDef.getType().getPackage()` it might be more accurate because names can be encoded.

Reply:
Yes `typeDef.getType().getPackage()` seems to be the better and more accurate approach.
Verified and fixed with dcd995921055c7108137082ab30f492f64f93996